### PR TITLE
[connectivity_for_web] Migration to null-safety.

### PR DESCRIPTION
**Pull request by ditman**
This change migrates the package to null safety, and moves its (very obsolete) tests to the new integration_test style.

The tests used to use mockito, but those mocks have been replaced by Fake implementations.

## Tests

* Tested manually with the connectivity/example app, in Sound null-safe mode.
* Tests still pass with minimal changes (rewritten Mockito mocks as Fakes)

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart [hunk 1]
```diff
@@ -9,12 +9,12 @@ class DartHtmlConnectivityPlugin extends ConnectivityPlugin {
...
   @override
   Future<ConnectivityResult> checkConnectivity() async {
-    return html.window.navigator.onLine
+    return html.window.navigator.onLine ?? false
         ? ConnectivityResult.wifi
         : ConnectivityResult.none;
   }
 
-  StreamController<ConnectivityResult> _connectivityResult;
+  late StreamController<ConnectivityResult> _connectivityResult;
```
Comment (stuartmorgan) [inline packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart:17]: Huh, I thought you couldn't null-check a `late` value (and that lazy initialization like the below needed `?` as a result).

File: packages/connectivity/connectivity_for_web/lib/src/network_information_api_connectivity_plugin.dart [hunk 2]
```diff
@@ -18,7 +18,7 @@ class NetworkInformationApiConnectivityPlugin extends ConnectivityPlugin {
 
   /// The constructor of the plugin.
   NetworkInformationApiConnectivityPlugin()
-      : this.withConnection(html.window.navigator.connection);
+      : this.withConnection(html.window.navigator.connection!);
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/connectivity/connectivity_for_web/lib/src/network_information_api_connectivity_plugin.dart:21]: Seems worth documenting that requirement in the constructor comment.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

File: packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart (hunk 1 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart:17]: You're right, let me give this a test!

**Comment by ditman** [comment]
The requirement of the 'connection' API is documented in the README: https://github.com/flutter/plugins/tree/master/packages/connectivity/connectivity_for_web#limitations-on-the-web-platform

File: packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart (hunk 1 above)
Comment (ditman) [reply packages/connectivity/connectivity_for_web/lib/src/dart_html_connectivity_plugin.dart]: This was indeed crashy. Fixed. This file needs a test (mock window + navigator), I'll add one after this lands.

**Pull request merged**